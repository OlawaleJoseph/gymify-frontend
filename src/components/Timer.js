/* eslint-disable consistent-return */
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

function Timer({ timeFrame }) {
  let interval;
  const [time, setTime] = useState('');
  const countDown = duration => {
    let timer = duration / 1000; let minutes; let seconds; let hours;
    interval = setInterval(() => {
      hours = Math.floor(parseInt(timer / 3600, 10));
      minutes = parseInt((timer % 3600) / 60, 10);
      seconds = parseInt(timer % 60, 10);

      hours = hours < 10 ? `0${hours}` : hours;
      minutes = minutes < 10 ? `0${minutes}` : minutes;
      seconds = seconds < 10 ? `0${seconds}` : seconds;
      setTime(`${hours} : ${minutes} : ${seconds}`);
      if (timer <= 0) clearInterval(interval);
      timer -= 1;
    }, 1000);
  };
  useEffect(() => {
    countDown(timeFrame);
    return () => clearInterval(interval);
  }, []);

  return (
    <p className="timer mb-0">{time}</p>
  );
}

Timer.propTypes = {
  timeFrame: PropTypes.number.isRequired,
};

export default Timer;
