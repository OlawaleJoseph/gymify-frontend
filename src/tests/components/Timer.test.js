import Timer from '../../components/Timer';
import { shallowRender } from '../setup';

describe('Side Bar', () => {
  const props = {
    timeFrame: 600,
  };
  const page = shallowRender(Timer, props);

  test('should render menu item', async () => {
    expect(page).toBeTruthy();
    expect(page).toMatchSnapshot();
  });
});
