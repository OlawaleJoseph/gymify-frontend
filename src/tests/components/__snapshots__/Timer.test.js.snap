// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Bar should render menu item 1`] = `ShallowWrapper {}`;
